packages/pipeline-editor/src/PipelineEditor/index.tsx

```diff
@@ -635,6 +636,76 @@ const PipelineEditor = forwardRef(
       return null;
     };
 
+    const getUpstreamNodeIds = (node: any): string[] => {
+      let upstreamNodeIds: string[] = [];
+
+      for (const link of node.inputs?.[0].links || []) {
+        upstreamNodeIds.push(link.node_id_ref);
+        const upstreamNode = controller.current.findExecutionNode(
+          link.node_id_ref
+        );
+        if (upstreamNode) {
+          upstreamNodeIds = upstreamNodeIds.concat(
+            getUpstreamNodeIds(upstreamNode)
+          );
+        }
+      }
+
+      return upstreamNodeIds;
+    };
```

Comment:
I will open an issue with canvas to see if it is possible to expose [this](https://github.com/elyra-ai/canvas/blob/48781597c137c4d5fc4222e5f19c1be87afca5f0/canvas_modules/common-canvas/src/object-model/object-model.js#L1869) or if its currently useable

Reply:
https://github.com/elyra-ai/canvas/issues/783

Reply:
@ajbozarth it looks like we have direct access to the objectModel via the canvas controller. As a temporary workaround, we should be able to use:
```js
this.objectModel.getHighlightObjectIds(pipelineID, nodeIDs, "upstream");
```

We should create a wrapper for this (similar to what you have here - `findUpstreamNodes(nodeID: string)`) and put it in our `PipelineController`.

I'm not sure if it makes more sense to return a list of node IDs or to return the actual nodes? I think we map everything to a real node eventually, so it might make sense to just do it upfront in this wrapper?